README.md

```diff
@@ -1,108 +1,119 @@
 ```
-curl -X POST http://midburn-tickets-queue.herokuapp.com/enqueue -d '{"firstname": "elad", "lastname": "gariany", "email":"email@gmail.com"}' --header "Content-Type: application/json"       
+bundle exec rake midburn:open_queue    # open the queue
+bundle exec rake midburn:close_queue   # close the queue
+bundle exec rake midburn:list          # Get List in CSV format
+bundle exec rake midburn:reset         # Reset the queue completely
 ```
 
-##### Params
-- `firstname` - Submitter's first name.
-- `lastname` - Submitter's last name.
-- `email` - Submitter's email address (profile on profile.midburn.org)
-
-### Known Issues
-- Orders in the system are limited to 500 * QUEUE_TIER_SIZE (say, 50,000 in case each tier size is 100).
-
-### Worker code
-```ruby
-
-def process_order(json = {"firstname":"elad","lastname":"gariany","email":"elad@gariany.com"})
-  # process order
-  # - validate with profile system, to confirm user have a profile
-  # - generate the email to be send to a client
-  # - send the email for purchase
-end
-
-class OrderTier_1
-  @queue = :tier_001
-  def self.perform(json)
-    process_order(JSON.parse(json))
-  end
-end
-```
+To get a full list of the available tasks, run `bundle exec rake -T midburn` and look for the
+
+### Deployment
+
+##### API Endpoints
+
+This is a simple [Sinatra](http://sinatrarb.com/) application. Please follow Sinatra's recommendation on how to deploy the app ([The Sinatra Book](http://sinatra-org-book.herokuapp.com/#toc_41)).
+
+For local development:
+- Make sure `bundle` is installed (if not follow [bundler.io](http://bundler.io/)).
+- Run `bundle install` to install dependencies.
+- Run server using: `bundle exec puma -p 3000 -C puma.rb`
+
+> **NOTICE**: use the `MIN_THREADS` & `MAX_THREADS` to set the number of puma threads to run on each api instance. For our need and since concurrency *was not* a concern with this simple solution & for our infrastructure, we found that each server may run very high (60+) amount of threads.
+
+##### Redis Server
+
+Queue implementation requires a running Redis server. Follow Redis instructions on how to install Redis instance [here](http://redis.io/topics/quickstart) (If you're on a mac and using [Homebrew](http://brew.sh), simply execute `brew install redis`).
+
+Once installed, configure the `REDIS_URL` env variable to your server (for example: `REDIS_URL="redis://localhost:6379/"`)
 
-## Configuration (Heroku)
-1. Close the queue:
-`heroku run bundle exec rake midburn:close_queue --app midburn-queue`
+##### Set Env Variables
 
-2. Open the queue:
-`heroku run bundle exec rake midburn:open_queue  --app midburn-queue`
+For local development, we use the [dotenv gem](https://github.com/bkeepers/dotenv). copy `.env-example` to `.env` and configure the variables for your need.
 
-3. Getting the list of emails in the queue
-`heroku run bundle exec rake midburn:list --app midburn-queue`
+For production needs, follow your environment's best practices.
```

Comment:
"... best practices for configuring env variables". For example, see Heroku's ["Configuration and Config Vars"](https://devcenter.heroku.com/articles/config-vars) & Amazon's EC2 ["Using Environment Variables"](http://docs.aws.amazon.com/opsworks/latest/userguide/apps-environment-vars.html).